dali/imgcodec/parsers/jpeg.cc

```diff
@@ -13,18 +13,68 @@
 // limitations under the License.
 
 #include "dali/imgcodec/parsers/jpeg.h"
+#include "dali/core/byte_io.h"
 
 namespace dali {
 namespace imgcodec {
 
+using jpeg_marker_t = std::array<uint8_t, 2>;
+
+constexpr jpeg_marker_t sos_marker = {0xff, 0xda};
+constexpr jpeg_marker_t soi_marker = {0xff, 0xd8};
+constexpr jpeg_marker_t eoi_marker = {0xff, 0xd9};
+
+bool IsValidMarker(const jpeg_marker_t &marker) {
+  return marker[0] == 0xff && marker[1] != 0x00;
+}
+
+bool IsSofMarker(const jpeg_marker_t &marker) {
+  if (!IsValidMarker(marker) || marker[1] < 0xc0 || marker[1] > 0xcf) return false;
+  return (marker[1] != 0xc4 && marker[1] != 0xc8 && marker[1] != 0xcc);
```

Comment:
Some comment would be nice.

Reply:
You're right, I added a comment in https://github.com/NVIDIA/DALI/pull/4050/commits/8dd065463c30d505584189d4f9ccaf787f021d64